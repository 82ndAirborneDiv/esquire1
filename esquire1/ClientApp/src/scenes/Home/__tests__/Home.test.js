import React from 'react'
import { shallow } from 'enzyme'
import { Home } from '../'

const props = {
  actions: {
    getProjectsRequest: jest.fn(),
    addProjectRequest: jest.fn()
  },
  projects: [
    { name: 'Project 1' },
    { name: 'Project 2' }
  ]
}

describe('Home scene', () => {
  test('it should render correctly', () => {
    expect(shallow(<Home {...props} />)).toMatchSnapshot()
  })
  
  test('it should handle ADD PROJECT button click event', () => {
    let wrapper = shallow(<Home {...props} />)
    wrapper.find('MainButton').simulate('click')
    expect(props.actions.addProjectRequest).toHaveBeenCalled()
  })
})