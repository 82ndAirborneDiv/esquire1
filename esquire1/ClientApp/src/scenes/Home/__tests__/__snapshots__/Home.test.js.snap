// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home scene it should render correctly 1`] = `
<withStyles(GridWrapper)
  container={true}
  direction="column"
>
  <withStyles(GridWrapper)
    item={true}
  >
    <h1>
      This is the home scene
    </h1>
  </withStyles(GridWrapper)>
  <withStyles(GridWrapper)
    item={true}
  >
    <h4>
      Pulled from SQL Server:
    </h4>
    <ul>
      <li
        key="Project 1"
      >
        Project 1
      </li>
      <li
        key="Project 2"
      >
        Project 2
      </li>
    </ul>
  </withStyles(GridWrapper)>
  <withStyles(GridWrapper)
    item={true}
  >
    <TextInput
      label="Project name"
      name="name"
      onChange={[Function]}
      placeholder="Enter project name"
      value=""
    />
    <MainButton
      color="primary"
      onClick={[Function]}
      value="Add project"
    />
  </withStyles(GridWrapper)>
</withStyles(GridWrapper)>
`;
